### PR TITLE
Add slash command attachment example

### PR DESCRIPTION
**Pull request by krittick**
## Summary

This adds an example for adding an attachment as a slash command option type.

This also fixes the missing `SlashCommandOptionType` reference in `__all__` in enums.py

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/enums.py [hunk 1]
```diff
@@ -61,6 +61,8 @@
     'ScheduledEventPrivacyLevel',
     'ScheduledEventLocationType',
     'InputTextStyle',
+    'SlashCommandOptionType',
+
```
Comment (Middledot) [inline discord/enums.py:65]:
```suggestion
```

**Review by Middledot — APPROVED** [review]

**Comment by VincentRPS** [comment]
@krittick why not just add this to the options example?

**Comment by krittick** [comment]
> @krittick why not just add this to the options example?

Since you have to do some additional processing to make it work (e.g. `to_file()`), I felt that it's better as a standalone example.

**Comment by VincentRPS** [comment]
> > @krittick why not just add this to the options example?
> 
> Since you have to do some additional processing to make it work (e.g. `to_file()`), I felt that it's better as a standalone example.

it's still a small enough example where it would be better there

**Comment by krittick** [comment]
> > > @krittick why not just add this to the options example?
> > 
> > 
> > Since you have to do some additional processing to make it work (e.g. `to_file()`), I felt that it's better as a standalone example.
> 
> it's still a small enough example where it would be better there

Sure, I've updated it now.

**Review by BobDotCom — DISMISSED** [review]

File: examples/app_commands/slash_options.py [hunk 2]
```diff
@@ -29,5 +31,18 @@ async def channel(
     await ctx.respond(f"Hi! You selected {channel.mention} channel.")
 
 
+@bot.slash_command(name="attach_file")
+async def say(
+    ctx: discord.ApplicationContext,
+    attachment: Option(
+        SlashCommandOptionType.attachment,
```
Comment (plun1331) [inline examples/app_commands/slash_options.py:38]: Using `discord.Attachment` here would be more consistent with the rest of the example

Comment (RGBCube) [reply examples/app_commands/slash_options.py:38]: i agree

File: examples/app_commands/slash_options.py [hunk 3]
```diff
@@ -29,5 +30,18 @@ async def channel(
     await ctx.respond(f"Hi! You selected {channel.mention} channel.")
 
 
+@bot.slash_command(name="attach_file")
+async def say(
+    ctx: discord.ApplicationContext,
+    attachment: Option(
+        discord.Attachment,
+        "File to attach to the message?",
```
Comment (Dorukyum) [inline examples/app_commands/slash_options.py:38]: This could be better maybe ```suggestion
        "The file to attach to the message",
```

**Review by BobDotCom — APPROVED** [review]

**Review by Dorukyum — APPROVED** [review]

**Pull request merged**